packages/trader/src/Stores/Modules/Trading/trade-store.js

```diff
@@ -1084,6 +1080,12 @@ export default class TradeStore extends BaseStore {
                 const { barrier_choices, max_stake, min_stake } = response.error.details;
                 this.setStakeBoundary(contract_type, min_stake, max_stake);
                 this.setStrikeChoices(barrier_choices);
+                if (duration_unit !== 'd' && !this.strike_price_choices.includes(this.barrier_1)) {
```

Comment:
Any specific reason you check if duration is not daily? Do we not need to set it for daily barrier as well?

Reply:
For duration in min & hours, change of either the type or duration value changes the list of barrier_choices. Hence went with this logic